src/java.base/share/native/libjli/java.c

```diff
@@ -660,6 +660,7 @@ JavaMain(void* _args)
         ret = 1;
     }
     LEAVE();
+    return ret;
```

Comment:
A `do-while` loop is always executed at least once, that is why the `LEAVE macro is structured as it is:
```
    do { \
        if ((*vm)->DetachCurrentThread(vm) != JNI_OK) { \
            JLI_ReportErrorMessage(JVM_ERROR2); \
            ret = 1; \
        } \
        if (JNI_TRUE) { \
            (*vm)->DestroyJavaVM(vm); \
            return ret; \
        } \
    } while (JNI_FALSE)
```
So this will always return and the sanitizer seems incorrect.

Reply:
Yes, I was make a mistake prior, the do-while(false) loop will execute once. But I think this PR still a bit usefull and no risk, the analyze seen as [below](https://github.com/openjdk/jdk/pull/22355#issuecomment-2497997812).